packages/devtools_app/test/cpu_profile_model_test.dart

```diff
@@ -265,6 +265,42 @@ void main() {
         equals(stackFrameD.exclusiveSampleCount),
       );
       expect(copy.inclusiveSampleCount, copy.exclusiveSampleCount);
+      expect(copy.sourceLine, equals(stackFrameD.sourceLine));
+    });
+
+    test('shallowCopy overrides', () {
+      final overrides = {
+        'id': 'overriddenId',
+        'name': 'overriddenName',
+        'verboseName': 'overriddenVerboseName',
+        'category': 'overriddenCategory',
+        'url': 'overriddenUrl',
+        'packageUri': 'overriddenPackageUri',
+        'parentId': 'overriddenParentId',
+      };
+      const overriddenSourceLine = 98329;
```

Comment:
what is the purpose of pulling this out as a special case? why not just include it in the overrides map?

Reply:
I would have needed to make the map values dynamic and didn't want to do any fancy casting